toolchain/check/testdata/impl/fail_self_type_mismatch.carbon

```diff
@@ -10,26 +10,25 @@
 
 class C[T:! type](X:! T) {}
 
-interface I {
-  // This tests that inside the definition of `I`, `Self` has type `I`.
-
-  // Below, `C(Self)` deduces `T` to be the incomplete facet type `I`, which
-  // leads to the error.
+// This tests that inside the definition of `I`, `Self` has type `I`. As a
+// result, in `impl T as I`, `Self` is `T as I`.
```

Comment:
I found this comment confusing: the problem is that name lookup for `Self` inside `impl T as I` finds `T` not `T as I`, which seems to be the opposite of what this comment is saying.

I think what you're getting at is that in `impl T as I`, the result of substituting into the `Self` used in `interface I` is `T as I`. (Which is not the same as the type `T` that `Self` names within `impl T as I`.)

Reply:
Updated comment. Better?